llumnix/llumlet/llumlet.py

```diff
@@ -98,30 +98,35 @@ def from_args(cls,
-            migrate_out_request = self.migration_scheduler.get_migrate_out_request()
             migrated_request_list = []
-            if migrate_out_request is None:
-                return migrated_request_list
-            status = self.migration_coordinator.migrate_out_multistage(migrate_in_ray_actor, migrate_out_request)
-            if status == MigrationStatus.FINISHED_DONE:
-                ray.get(migrate_in_ray_actor.execute_engine_method.remote("commit_dst_request", migrate_out_request))
-                self.backend_engine.free_src_request(migrate_out_request)
-                migrated_request_list.append(migrate_out_request.request_id)
-                migrate_out_request.stage_timestamps.append(time.time())
-                self.backend_engine.remove_migrating_out_request_last_stage(migrate_out_request)
-            else:
-                migrate_out_request.reset_migration_args()
-                ray.get(migrate_in_ray_actor.execute_migration_method.remote("free_dst_pre_alloc_cache", migrate_out_request.request_id))
-            t1 = time.time()
-            logger.info("{}->{} migrate done, migrate request {}, status:{}, len:{} blocks, cost:{} ms" \
-                  .format(self.instance_id, dst_instance_id, migrated_request_list, status, \
-                   sum(migrate_out_request.stage_num_blocks_list), (t1 - t0)*1000))
+            continue_migrate = True
+            while continue_migrate and (len(migrated_request_list) < num_requests or num_requests == -1):
+                t0 = time.time()
+                migrate_out_request = self.migration_scheduler.get_migrate_out_request()
+                if migrate_out_request is not None:
+                    logger.info("migrate_out {}".format(migrate_out_request.request_id))
+                if migrate_out_request is None:
+                    return migrated_request_list
+                logger.info("{}->{} begin migrate out {}".format(self.instance_id, dst_instance_id, migrate_out_request.request_id))
+                status = self.migration_coordinator.migrate_out_multistage(migrate_in_ray_actor, migrate_out_request)
+                if status == MigrationStatus.FINISHED_DONE:
+                    ray.get(migrate_in_ray_actor.execute_engine_method.remote("commit_dst_request", migrate_out_request))
+                    self.backend_engine.free_src_request(migrate_out_request)
+                    migrated_request_list.append(migrate_out_request.request_id)
+                    migrate_out_request.stage_timestamps.append(time.time())
+                    self.backend_engine.remove_migrating_out_request_last_stage(migrate_out_request)
+                else:
+                    migrate_out_request.reset_migration_args()
+                    ray.get(migrate_in_ray_actor.execute_migration_method.remote("free_dst_pre_alloc_cache", migrate_out_request.request_id))
+                    continue_migrate = False
```

Comment:
If migration aborted by request itself, we should continue migrate other request.

Reply:
> If migration aborted by request itself, we should continue migrate other request.

The original logic does not continue new migrations when migration aborted. Please confirm again whether we need to modify this logic.  Moreover, if there isn't enough space on the dst, we shouldn't try to migrate new requests in a loop.

Reply:
If a request is done during migration, its status is also `MIgrationStatus.FINISHED_ABORTED`, in that case, we should continue migrate.  You can add a new `MigrationStatus` to distinguish these cases.